daemon/algod/api/server/v2/handlers.go

```diff
@@ -447,3 +452,24 @@ func (v2 *Handlers) GetPendingTransactions(ctx echo.Context, params generated.Ge
 func (v2 *Handlers) GetPendingTransactionsByAddress(ctx echo.Context, addr string, params generated.GetPendingTransactionsByAddressParams) error {
 	return v2.getPendingTransactions(ctx, params.Max, params.Format, &addr)
 }
+
+// Compile TEAL code to binary, return both binary and hash
+// (POST /v2/compile)
+func (v2 *Handlers) TealCompile(ctx echo.Context) error {
+	buf := new(bytes.Buffer)
+	ctx.Request().Body = http.MaxBytesReader(nil, ctx.Request().Body, maxTealSourceBytes)
+	buf.ReadFrom(ctx.Request().Body)
+	source := buf.String()
+	program, err := logic.AssembleString(source)
+	if err != nil {
+		return badRequest(ctx, err, err.Error(), v2.Log)
+	}
+	
+	pd := logic.HashProgram(program)
+	addr := basics.Address(pd)
+	response := generated.PostCompileResponse {
+		Hash: addr.String(),
```

Comment:
it's spitting out the hash with a checksum - is that intentional?

Reply:
it should be Address style because if the program is for logic-owned stateless teal then the hash of the program is the account Address